actionpack/lib/action_dispatch/system_testing/test_helpers/screenshot_helper.rb

```diff
@@ -34,54 +34,70 @@ def take_screenshot
         # fails add +take_failed_screenshot+ to the teardown block before clearing
         # sessions.
         def take_failed_screenshot
+          @screenshot_unique_suffix = nil
           take_screenshot if failed? && supports_screenshot?
         end
 
         private
-          def image_name
-            failed? ? "failures_#{method_name}" : method_name
+
+          def save_image
+            save_image_with_available_suffix
+          rescue Errno::ENAMETOOLONG
+            truncate_filename_base
+            save_image_with_available_suffix
           end
 
-          def image_path
-            @image_path ||= absolute_image_path.to_s
+          def save_image_with_available_suffix
+            increment_unique_suffix while filename_taken?
+            save_image!
           end
 
-          def absolute_image_path
-            Rails.root.join("tmp/screenshots/#{image_name}.png")
+          def save_image!
+            page.save_screenshot(image_path)
           end
 
-          def save_image
-            page.save_screenshot(absolute_image_path)
+          def filename_taken?
+            image_path.exist?
           end
 
-          def output_type
-            # Environment variables have priority
-            output_type = ENV["RAILS_SYSTEM_TESTING_SCREENSHOT"] || ENV["CAPYBARA_INLINE_SCREENSHOT"]
+          def image_path
+            Rails.root.join("tmp/screenshots", filename)
+          end
+
+          def filename
+            @screenshot_unique_suffix ||= nil # avoid "not initialized" warning
+            prefix = "failures_" if failed? # else nil
+            "#{prefix}#{filename_base}#{@screenshot_unique_suffix}.png"
+          end
+
+          def filename_base
+            @screenshot_filename_base ||= method_name
+          end
 
-            # Default to outputting a path to the screenshot
-            output_type ||= "simple"
```

Comment:
I don't see this logic anymore, is that intended?

Reply:
I'm new to open source, so I apologize if I overstepped what I should change. It seemed the `"simple"` value wasn't actually used for anything; only the special values `"artifact"` and `"inline"` were used. I'm happy to put "simple" back as the default.